src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -23,13 +23,26 @@ public static class Program
         public static int Main(string[] args)
         {
             const int expectedArguments = 17;
+            const int expectedArgumentsWithValidateOnly = 18; // Extra --validate-only argument for CI checks.
 
-            if (args.Length != expectedArguments)
+            if (args.Length is not (expectedArguments or expectedArgumentsWithValidateOnly))
             {
-                Console.Error.WriteLine($"Excepted {expectedArguments} arguments, found {args.Length}: {string.Join(';', args)}");
+                Console.Error.WriteLine($"Excepted {expectedArguments} or {expectedArgumentsWithValidateOnly} arguments, found {args.Length}: {string.Join(';', args)}");
```

Comment:
Instead of showing simply the 2 expected args count numbers it might be good to explicitly name the extra arg, don't you think?